### PR TITLE
[fluorine] Fixes to scheduler for jobs with seconds, minutes, etc.

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Do not log about various things like jid_include and max_running unless the job is set to run.  If the job contains _seconds, eg. it's using seconds, minutes, hours, etc. then only set the _next_fire_time if the job actually runs.  Adding more tests to test simple scenarios when seconds, minutes, hours and days to ensure we get multiple scheduled job runs when they're expected to run.

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/49082
https://github.com/saltstack/salt/issues/49093

### Tests written?
Yes

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/utils/schedule.py
Comment (pizzapanther) [reply salt/utils/schedule.py]: I think I worked around this so I don't think it will break the enterprise scheduler but it does mean we could probably simplify some code now. I'll test it out today and report back.

**Comment by cachedout** [comment]
Please don't merge this until we have signoffs from both @s0undt3ch and @pizzapanther

**Review by s0undt3ch — COMMENTED** [review]
I defer my review to @pizzapanther

**Review by pizzapanther — CHANGES_REQUESTED** [review]
A couple findings. First in enterprise we use salt as a python import when calculating schedules and there looks to be a loading issue:

```
File "/root/.local/share/virtualenvs/raas-xhpEmuIf/lib/python3.6/site-packages/salt/utils/schedule.py", line 101, in __new__
    utils=utils)
  File "/root/.local/share/virtualenvs/raas-xhpEmuIf/lib/python3.6/site-packages/salt/utils/schedule.py", line 132, in __singleton_init__
    self.utils = utils or salt.loader.utils(opts)
  File "/root/.local/share/virtualenvs/raas-xhpEmuIf/lib/python3.6/site-packages/salt/loader.py", line 339, in utils
    _module_dirs(opts, 'utils', ext_type_dirs='utils_dirs'),
  File "/root/.local/share/virtualenvs/raas-xhpEmuIf/lib/python3.6/site-packages/salt/loader.py", line 147, in _module_dirs
    ext_types = os.path.join(opts['extension_modules'], ext_type)
KeyError: 'extension_modules'
```
My guess is you are loading `extension_modules` when salt gets executed as a CLI command and not doing it when used as a python import. So should we file a separate bug for that?

Anyways I couldn't get it to work right with out a little hacking. But once I worked around the loader and got it calculating schedules and there are big differences.

I'm thinking maybe we submit some of the enterprise code for a unit test on the scheduler so you can see how we use it. I put some work arounds in to use it for our use case. Maybe we need to use it more "properly" or also this change might have broken the calculation too much.

**Comment by garethgreenaway** [comment]
@pizzapanther For the `extension_modules` issue, a quick fix would be to add that to the `extension_modules` to the `opts` that you're passing to salt.utils.schedule.Schedule.

**Comment by pizzapanther** [comment]
adding `extension_modules` worked, thanks! Maybe it should default to an empty string though?

**Review by pizzapanther — APPROVED** [review]
OK this works for me now.

**Comment by garethgreenaway** [comment]
Awesome!  I have a couple more tests I want to add before merging just to make sure everything is good. Thanks @pizzapanther.

**Comment by cachedout** [comment]
@garethgreenaway Are you ready to go here?

**Comment by garethgreenaway** [comment]
@cachedout yup!  Good to go on my end.

**Review by cachedout — APPROVED** [review]

**Pull request merged**
